services/src/api/model/datatypes.rs

```diff
@@ -1719,13 +1720,122 @@ impl From<RasterPropertiesEntryType> for geoengine_datatypes::raster::RasterProp
     }
 }
 
-#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Debug, ToSchema)]
+#[derive(PartialEq, Eq, Clone, Debug)]
 pub struct DateTimeParseFormat {
     fmt: String,
     has_tz: bool,
     has_time: bool,
 }
 
+enum FormatStrLoopState {
+    Normal,
+    Percent(String),
+}
+
+impl DateTimeParseFormat {
+    pub fn custom(fmt: String) -> Self {
```

Comment:
This looks rather complicated. Maybe test it?

Reply:
Yeah, back then it was tested and then I copied it doesn't sound good.
Rather copy the test or do the conversion.

Reply:
5b5c0e77ea24da8f4adb41af2be807332aa341ef